File: LinearSearch.cpp
#include <bits/stdc++.h>

using namespace std;
/*
 * Complete the function below.
 */
string findNumber(vector < int > arr, int k) {
    auto it = std::find(arr.begin(),arr.end(),k);
    if(it!= arr.end())
        return "YES";
    else
        return "NO";

}

int main() {
    ofstream fout(getenv("OUTPUT_PATH"));
    string res;
    
    int _arr_size = 0;
    cin >> _arr_size;
    cin.ignore (std::numeric_limits<std::streamsize>::max(), '\n'); 
    vector<int> _arr;
    int _arr_item;
    for(int _arr_i=0; _arr_i<_arr_size; _arr_i++) {
        cin >> _arr_item;
        cin.ignore (std::numeric_limits<std::streamsize>::max(), '\n');
        _arr.push_back(_arr_item);
    }
    
    int _k;
    cin >> _k;
    cin.ignore (std::numeric_limits<std::streamsize>::max(), '\n');
    
    res = findNumber(_arr, _k);
    fout << res << endl;
    
    fout.close();
    return 0;
}

File: LinearSearch.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;

public class Solution {

 /*
     * Complete the findNumber function below.
     */
    static String findNumber(int[] arr, int k) {
        /*
         * Write your code here.
         */
            
         boolean found = Arrays.stream(arr).anyMatch(x -> x == k);
         if(found)
             return "YES";
        else
            return "NO";
    }




 private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int arrCount = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])*");

        int[] arr = new int[arrCount];

        for (int arrItr = 0; arrItr < arrCount; arrItr++) {
            int arrItem = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])*");
            arr[arrItr] = arrItem;
        }

        int k = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])*");

        String res = findNumber(arr, k);

        bufferedWriter.write(res);
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}

